rasa/core/policies/embedding_policy.py

```diff
@@ -563,10 +563,10 @@ def predict_action_probabilities(
         confidence = confidence[0, -1, :].tolist()
 
         # normalise scores if turned on
-        if self.loss_type == "softmax" and self.normalize_top_num_actions > 0:
+        if self.loss_type == "softmax" and self.ranking_length > 0:
             ranked = sorted(confidence, reverse=True)
             for i, value in enumerate(confidence):
-                if value < ranked[self.normalize_top_num_actions - 1]:
+                if value < ranked[self.ranking_length - 1]:
                     confidence[i] = 0.0
```

Comment:
if you keep confidence as numpy array, then simply `confidence[confidence < ranked[self.ranking_length - 1]] = 0` is enough

Reply:
aha, thanks. was trying to use `np.where()` to that affect but couldn't quite figure it out